File: src/main/scala/com/yogpc/qp/machines/base/IDisabled.scala
package com.yogpc.qp.machines.base

import com.yogpc.qp.Config

trait IDisabled {

  def getSymbol: Symbol

  def enabled: Boolean = Config.common.disabled.get(getSymbol).exists(!_.get().booleanValue())

  def defaultDisableMachine = false
}

File: src/main/scala/com/yogpc/qp/modules/IDisabled.java
package com.yogpc.qp.modules;

import com.yogpc.qp.Config;

public interface IDisabled {
    scala.Symbol getSymbol();

    default boolean enabled() {
        return Config.content().disableMapJ().getOrDefault(getSymbol(), true);
    }

    default boolean defaultDisableMachine() {
        return false;
    }
}
